driver-core/src/main/com/mongodb/internal/dns/DefaultDnsResolver.java

```diff
@@ -39,6 +35,23 @@
  */
 public final class DefaultDnsResolver implements DnsResolver {
 
+    private final DnsClient dnsClient;
+
+    public DefaultDnsResolver() {
+        ServiceLoader<DnsClientProvider> loader = ServiceLoader.load(DnsClientProvider.class);
```

Comment:
```suggestion
        ServiceLoader<DnsClientProvider> loader = ServiceLoader.load(DnsClientProvider.class, Thead.currentThread().getContextClassloader());
```

Reply:
In the ServiceLoader javadoc it says that these two things are equivalent.